lib/dao.js

```diff
@@ -2742,6 +2748,7 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
             inst.__persisted = true;
             inst.setAttributes(ctx.data);
+            inst[pkName] = id;
```

Comment:
Can we please reset the PK column in the data, before it's set in the model instance?

```js
ctx.data[pkName] = id;
inst.setAttributes(ctx.data);
```

Should we detect the case when a "loaded" hook is trying to change PK field and report a warning too?